### PR TITLE
Redshift crawler index of out bound issue [sc-6687]

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

When parsing precision, max_length from `fotmat_type` could cause `IndexError` for some reason. According to the document, `character varying` and `character` should be at least 1.

### 🤓 What?

- Add try-catch to safely parse `format_type`.

### 🧪 Tested?

n/a

### CODE REVIEW

File: metaphor/postgresql/extractor.py [hunk 1]
```diff
@@ -306,10 +306,16 @@ def parse_format_type(
             elif native_type == "double precision":
                 precision = 53
```
Comment (alyiwang) [inline metaphor/postgresql/extractor.py:307]: Do we need to make `24` and `53` as `24.0` and `53.0`, validation may complain about casting int to float

File: metaphor/postgresql/extractor.py [hunk 2]
```diff
@@ -306,10 +306,16 @@ def parse_format_type(
...
+                    precision = float(type_str.split("(")[1].split(",")[0])
+                except IndexError:
+                    logger.warning(f"Failed to parse precision from {type_str}.")
 
             if native_type == "character varying" or native_type == "character":
-                max_length = float(type_str.split("(")[1].strip(")"))
+                try:
+                    max_length = float(type_str.split("(")[1].strip(")"))
+                except IndexError:
+                    logger.warning(f"Failed to parse max_length from {type_str}.")
```
Comment (alyiwang) [inline metaphor/postgresql/extractor.py:318]: can we add a default `else` statement, simply logging the unsupported cases

**Review by alyiwang — COMMENTED** [review]
Do we have an example of `type_str` value that failed the parsing?

File: metaphor/postgresql/extractor.py (hunk 2 above)
Comment (elic-eon) [reply metaphor/postgresql/extractor.py:318]: Do you mean to catch other exceptions?

File: metaphor/postgresql/extractor.py [hunk 3]
```diff
@@ -302,14 +302,20 @@ def parse_format_type(
...
                 precision = 64.0
             elif native_type == "real":
-                precision = 24
+                precision = 24.0
             elif native_type == "double precision":
-                precision = 53
+                precision = 53.0
             elif native_type == "numeric" and type_str != "numeric":
-                precision = float(type_str.split("(")[1].split(",")[0])
+                try:
```
Comment (mars-lan) [inline metaphor/postgresql/extractor.py:309]: Let's move this try-catch block to a util method to share code between `numeric` & `char` types.

**Pull request merged**